front/app/modules/commercial/customizable_homepage_banner/admin/CallToActionButtonSettings.tsx

```diff
@@ -14,39 +14,62 @@ import {
 } from 'services/appConfiguration';
 
 interface SigninStateProps {
-  options: string[];
+  options: (CTASignedOutButton | CTASignedInButton)[];
```

Comment:
You can make [conditional props](https://www.benmvp.com/blog/conditional-react-props-typescript/). So if signInState is a certain value, you want CTASignedOutButton[], otherwise you want CTASignedInButton[]

Reply:
Great hint! Thank you!